<a href="https://colab.research.google.com/github/hbabar1215/nhanes_inferential_2021_23/blob/main/nhanes_inferential_2021_23_R.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Install R packages needed
install.packages("ggplot2")
install.packages("lmtest")
install.packages("readr")
install.packages("haven")
install.packages("dplyr")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependency ‘zoo’


Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [2]:
# Load the packages
library(ggplot2)
library(lmtest)
library(readr)
library(haven)
library(dplyr)

Loading required package: zoo


Attaching package: ‘zoo’


The following objects are masked from ‘package:base’:

    as.Date, as.Date.numeric



Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union




In [3]:
# Question 1: "Is there an association between marital status (married or not married) and education level (bachelor’s degree or higher vs. less than a bachelor’s degree)?"

In [4]:
# Load Demographic data
df_demo <- read_xpt("/content/DEMO_L.xpt")
head(df_demo)

SEQN   SDDSRVYR RIDSTATR RIAGENDR RIDAGEYR RIDAGEMN RIDRETH1 RIDRETH3
1 130378 12       2        1        43       NA       5        6       
2 130379 12       2        1        66       NA       3        3       
3 130380 12       2        2        44       NA       2        2       
4 130381 12       2        2         5       NA       5        7       
5 130382 12       2        1         2       NA       3        3       
6 130383 12       1        2         3       NA       2        2       
  RIDEXMON RIDEXAGM ⋯ DMDHRGND DMDHRAGZ DMDHREDZ DMDHRMAZ DMDHSEDZ WTINT2YR
1  2       NA       ⋯ NA       NA       NA       NA       NA       50055.45
2  2       NA       ⋯ NA       NA       NA       NA       NA       29087.45
3  1       NA       ⋯ NA       NA       NA       NA       NA       80062.67
4  1       71       ⋯  2        2        2        3       NA       38807.27
5  2       34       ⋯  2        2        3        1        2       30607.52
6 NA       NA       ⋯  2        2       NA        1       NA       43456.14
  WTMEC2YR SDMVSTRA SDMVPSU INDFMPIR
1 54374.46 173      2       5.00    
2 34084.72 173      2       5.00    
3 81196.28 174      1       1.41    
4 55698.61 182      2       1.53    
5 36434.15 182      2       3.60    
6     0.00 176      2         NA

In [5]:
# Keep columns DMDMARTZ and DMDEDUC2
df_demo_new<- df_demo[, c("DMDMARTZ", "SEQN", "DMDEDUC2")]
head(df_demo_new)

DMDMARTZ SEQN   DMDEDUC2
1  1       130378  5      
2  1       130379  5      
3  1       130380  3      
4 NA       130381 NA      
5 NA       130382 NA      
6 NA       130383 NA

In [6]:
# Drop missing data or invalid codes
df_clean <- na.omit(df_demo_new)
head(df_clean)

DMDMARTZ SEQN   DMDEDUC2
1 1        130378 5       
2 1        130379 5       
3 1        130380 3       
4 3        130384 2       
5 1        130385 3       
6 1        130386 4

In [7]:
# Recode education level
df_clean <-df_clean %>%
  mutate(
    education_level = ifelse(DMDEDUC2 == 5, "Bachelor's or Higher", "Less than Bachelor's")
  )
  head(df_clean)

DMDMARTZ SEQN   DMDEDUC2 education_level     
1 1        130378 5        Bachelor's or Higher
2 1        130379 5        Bachelor's or Higher
3 1        130380 3        Less than Bachelor's
4 3        130384 2        Less than Bachelor's
5 1        130385 3        Less than Bachelor's
6 1        130386 4        Less than Bachelor's

In [8]:
# Recode marital status
df_clean <-df_clean %>%
  mutate(
    marital_status = ifelse(DMDMARTZ== 1, "Married", "Not Married")
  )
  head(df_clean)

DMDMARTZ SEQN   DMDEDUC2 education_level      marital_status
1 1        130378 5        Bachelor's or Higher Married       
2 1        130379 5        Bachelor's or Higher Married       
3 1        130380 3        Less than Bachelor's Married       
4 3        130384 2        Less than Bachelor's Not Married   
5 1        130385 3        Less than Bachelor's Married       
6 1        130386 4        Less than Bachelor's Married

In [9]:
colnames(df_demo_new)

[1] "DMDMARTZ" "SEQN"     "DMDEDUC2"

In [10]:
  # Keep only the new recoded columns
  df_demo_clean <- df_clean %>%
  select(SEQN, marital_status, education_level) %>%
  filter(!is.na(marital_status) & !is.na(education_level))

  head(df_demo_clean)


SEQN   marital_status education_level     
1 130378 Married        Bachelor's or Higher
2 130379 Married        Bachelor's or Higher
3 130380 Married        Less than Bachelor's
4 130384 Not Married    Less than Bachelor's
5 130385 Married        Less than Bachelor's
6 130386 Married        Less than Bachelor's

In [11]:
# Create table
table_question1 <- table(df_clean$marital_status, df_clean$education_level)
print(table_question1)

             
              Bachelor's or Higher Less than Bachelor's
  Married                     1631                 2505
  Not Married                  994                 2662


In [12]:
# Create contigency table
chi_result <- chisq.test(table_question1)
chi_result


	Pearson's Chi-squared test with Yates' continuity correction

data:  table_question1
X-squared = 129.73, df = 1, p-value < 2.2e-16


In [13]:
cat("Chi-square test p-value:", chi_result$p.value, "\n")

Chi-square test p-value: 4.701728e-30 


In [14]:
# Pritn result
if (chi_result$p.value < 0.05) {
  cat("There is a significant relationship between marital status and education level.\n")
} else {
  cat("There is no significant relationship between marital status and education level.\n")
}

There is a significant relationship between marital status and education level.


Answer: This suggests that there is an association between marital level and education level. People who are married are more likely to have a bachelor's degree or higher.

In [15]:
# Question 2: Is there a difference in the mean sedentary behavior time between those who are married and those who are not married?

In [16]:
# Load PAQ data
df_paq <- read_xpt("/content/PAQ_L.xpt")
df_paq

SEQN   PAD790Q PAD790U PAD800 PAD810Q PAD810U PAD820 PAD680
1    130378 3       W        45    3       W       45      360  
2    130379 4       W        45    3       W       45      480  
3    130380 1       W        20    0               NA      240  
4    130384 0                NA    0               NA       60  
5    130385 1       D        90    1       W       60      180  
6    130386 1       W        30    1       M       30      180  
7    130387 0                NA    0               NA     1200  
8    130388 5       W       180    0               NA      360  
9    130389 3       W        45    2       M       30      720  
10   130390 1       W        60    0               NA      300  
11   130391 3       D        10    1       D       20      240  
12   130392 0                NA    0               NA      300  
13   130393 4       W        45    1       W       60      720  
14   130394 7       W        45    4       W       30      420  
15   130395 3       W        15    0               NA      360  
16   130396 3       W        20    0               NA      240  
17   130397 0                NA    0               NA      480  
18   130398 7       W       120    2       Y       60       60  
19   130399 6       W        90    2       W       60      360  
20   130400 0                NA    0               NA      960  
21   130401 0                NA    0               NA      960  
22   130402 1       W        60    1       M       60      840  
23   130404 5       W        50    3       W       30      120  
24   130406 5       W        60    3       W       60      300  
25   130407 5       W        50    1       W       40      180  
26   130408 1       D        45    3       W       45      120  
27   130409 3       W        60    1       M       60      300  
28   130410 0                NA    0               NA      120  
29   130413 5       W        50    0               NA       60  
30   130415 2       W       120    0               NA      360  
⋮    ⋮      ⋮       ⋮       ⋮      ⋮       ⋮       ⋮      ⋮     
8124 142268    6    W       360    1       W       120    300   
8125 142269    0             NA    0                NA    600   
8126 142270    0             NA    0                NA    960   
8127 142273    5    W        60    5       W        60    420   
8128 142278    1    D        30    0                NA    180   
8129 142279    2    W       150    0                NA    420   
8130 142280    1    W       120    1       W        60    180   
8131 142283 9999             NA    2       W        60    300   
8132 142284    1    D        60    3       W        60    360   
8133 142285    3    W        30    0                NA    300   
8134 142286    3    W       180    0                NA    420   
8135 142287    2    D        60    0                NA    300   
8136 142288    7    W        30    0                NA    360   
8137 142289    3    W        60    3       W        60    150   
8138 142290    1    D        60    2       W        60    240   
8139 142293    2    W        15    0                NA    300   
8140 142294    0             NA    0                NA    960   
8141 142295    0             NA    0                NA    480   
8142 142297    3    W        15    5       M        45    300   
8143 142298    0             NA    0                NA    480   
8144 142299    1    D        60    1       D        30    120   
8145 142300    0             NA    0                NA     90   
8146 142301    0             NA    0                NA    360   
8147 142302    1    D        45    1       D        20    120   
8148 142303    2    D        60    0                NA    360   
8149 142305    2    W        40    0                NA    480   
8150 142307    3    W        15    0                NA    480   
8151 142308    1    W        45    0                NA    600   
8152 142309    2    D        15    0                NA    240   
8153 142310    0             NA    0    

In [17]:
# Merge the demographic data and PAQ data
df_merged2 <- merge(df_demo_clean, df_paq, by = "SEQN")
df_merged2

SEQN   marital_status education_level      PAD790Q PAD790U PAD800 PAD810Q
1    130378 Married        Bachelor's or Higher 3       W        45    3      
2    130379 Married        Bachelor's or Higher 4       W        45    3      
3    130380 Married        Less than Bachelor's 1       W        20    0      
4    130384 Not Married    Less than Bachelor's 0                NA    0      
5    130385 Married        Less than Bachelor's 1       D        90    1      
6    130386 Married        Less than Bachelor's 1       W        30    1      
7    130387 Not Married    Bachelor's or Higher 0                NA    0      
8    130388 Married        Less than Bachelor's 5       W       180    0      
9    130389 Married        Bachelor's or Higher 3       W        45    2      
10   130390 Married        Less than Bachelor's 1       W        60    0      
11   130391 Not Married    Less than Bachelor's 3       D        10    1      
12   130392 Married        Bachelor's or Higher 0                NA    0      
13   130393 Married        Bachelor's or Higher 4       W        45    1      
14   130394 Married        Bachelor's or Higher 7       W        45    4      
15   130395 Not Married    Less than Bachelor's 3       W        15    0      
16   130396 Married        Less than Bachelor's 3       W        20    0      
17   130397 Married        Less than Bachelor's 0                NA    0      
18   130398 Married        Less than Bachelor's 7       W       120    2      
19   130399 Married        Bachelor's or Higher 6       W        90    2      
20   130400 Not Married    Less than Bachelor's 0                NA    0      
21   130402 Not Married    Bachelor's or Higher 1       W        60    1      
22   130404 Not Married    Less than Bachelor's 5       W        50    3      
23   130406 Married        Less than Bachelor's 5       W        60    3      
24   130407 Not Married    Bachelor's or Higher 5       W        50    1      
25   130408 Not Married    Less than Bachelor's 1       D        45    3      
26   130409 Married        Less than Bachelor's 3       W        60    1      
27   130410 Not Married    Less than Bachelor's 0                NA    0      
28   130413 Married        Less than Bachelor's 5       W        50    0      
29   130415 Married        Less than Bachelor's 2       W       120    0      
30   130416 Not Married    Less than Bachelor's 0                NA    0      
⋮    ⋮      ⋮              ⋮                    ⋮       ⋮       ⋮      ⋮      
7763 142267 Married        Bachelor's or Higher    4    W        30    3      
7764 142268 Married        Less than Bachelor's    6    W       360    1      
7765 142269 Not Married    Less than Bachelor's    0             NA    0      
7766 142270 Married        Less than Bachelor's    0             NA    0      
7767 142278 Not Married    Less than Bachelor's    1    D        30    0      
7768 142279 Married        Less than Bachelor's    2    W       150    0      
7769 142280 Not Married    Less than Bachelor's    1    W       120    1      
7770 142283 Married        Less than Bachelor's 9999             NA    2      
7771 142284 Married        Bachelor's or Higher    1    D        60    3      
7772 142285 Married        Less than Bachelor's    3    W        30    0      
7773 142286 Married        Less than Bachelor's    3    W       180    0      
7774 142287 Married        Bachelor's or Higher    2    D        60    0      
7775 142288 Married        Less than Bachelor's    7    W        30    0      
7776 142289 Married        Bachelor's or Higher    3    W        60    3      
7777 142290 Married        Less than Bachelor's    1    D        60    2      
7778 142293 Married        Less than Bachelor's    2    W        15    0      
7779 142294 Not Married    Less than Bachelor's    0             NA    0      
7780 142295 Not Married    Less than Bachelor's    0             NA    0      
7781 142297 Not Married    Less than Bachelor's    3   

In [18]:
# Keep only SEQN, marital_status, and PAD680 columns
df_question2 <- df_merged2[, c("SEQN", "marital_status", "PAD680")]
df_question2

SEQN   marital_status PAD680
1    130378 Married         360  
2    130379 Married         480  
3    130380 Married         240  
4    130384 Not Married      60  
5    130385 Married         180  
6    130386 Married         180  
7    130387 Not Married    1200  
8    130388 Married         360  
9    130389 Married         720  
10   130390 Married         300  
11   130391 Not Married     240  
12   130392 Married         300  
13   130393 Married         720  
14   130394 Married         420  
15   130395 Not Married     360  
16   130396 Married         240  
17   130397 Married         480  
18   130398 Married          60  
19   130399 Married         360  
20   130400 Not Married     960  
21   130402 Not Married     840  
22   130404 Not Married     120  
23   130406 Married         300  
24   130407 Not Married     180  
25   130408 Not Married     120  
26   130409 Married         300  
27   130410 Not Married     120  
28   130413 Married          60  
29   130415 Married         360  
30   130416 Not Married     180  
⋮    ⋮      ⋮              ⋮     
7763 142267 Married        480   
7764 142268 Married        300   
7765 142269 Not Married    600   
7766 142270 Married        960   
7767 142278 Not Married    180   
7768 142279 Married        420   
7769 142280 Not Married    180   
7770 142283 Married        300   
7771 142284 Married        360   
7772 142285 Married        300   
7773 142286 Married        420   
7774 142287 Married        300   
7775 142288 Married        360   
7776 142289 Married        150   
7777 142290 Married        240   
7778 142293 Married        300   
7779 142294 Not Married    960   
7780 142295 Not Married    480   
7781 142297 Not Married    300   
7782 142298 Married        480   
7783 142299 Married        120   
7784 142300 Married         90   
7785 142301 Not Married    360   
7786 142302 Not Married    120   
7787 142303 Not Married    360   
7788 142305 Not Married    480   
7789 142307 Not Married    480   
7790 142308 Married        600   
7791 142309 Not Married    240   
7792 142310 Married        360

In [19]:
# Remove rows where PAD680 is 7777, 9999, or NA
df_question2 <- df_question2[!(df_question2$PAD680 %in% c(7777, 9999)) & !is.na(df_question2$PAD680), ]
df_question2

SEQN   marital_status PAD680
1    130378 Married         360  
2    130379 Married         480  
3    130380 Married         240  
4    130384 Not Married      60  
5    130385 Married         180  
6    130386 Married         180  
7    130387 Not Married    1200  
8    130388 Married         360  
9    130389 Married         720  
10   130390 Married         300  
11   130391 Not Married     240  
12   130392 Married         300  
13   130393 Married         720  
14   130394 Married         420  
15   130395 Not Married     360  
16   130396 Married         240  
17   130397 Married         480  
18   130398 Married          60  
19   130399 Married         360  
20   130400 Not Married     960  
21   130402 Not Married     840  
22   130404 Not Married     120  
23   130406 Married         300  
24   130407 Not Married     180  
25   130408 Not Married     120  
26   130409 Married         300  
27   130410 Not Married     120  
28   130413 Married          60  
29   130415 Married         360  
30   130416 Not Married     180  
⋮    ⋮      ⋮              ⋮     
7763 142267 Married        480   
7764 142268 Married        300   
7765 142269 Not Married    600   
7766 142270 Married        960   
7767 142278 Not Married    180   
7768 142279 Married        420   
7769 142280 Not Married    180   
7770 142283 Married        300   
7771 142284 Married        360   
7772 142285 Married        300   
7773 142286 Married        420   
7774 142287 Married        300   
7775 142288 Married        360   
7776 142289 Married        150   
7777 142290 Married        240   
7778 142293 Married        300   
7779 142294 Not Married    960   
7780 142295 Not Married    480   
7781 142297 Not Married    300   
7782 142298 Married        480   
7783 142299 Married        120   
7784 142300 Married         90   
7785 142301 Not Married    360   
7786 142302 Not Married    120   
7787 142303 Not Married    360   
7788 142305 Not Married    480   
7789 142307 Not Married    480   
7790 142308 Married        600   
7791 142309 Not Married    240   
7792 142310 Married        360

In [20]:
# Calculate mean PAD680 by marital status
married_data <- subset(df_question2, marital_status == "Married")$PAD680
not_married_data <- subset(df_question2, marital_status == "Not Married")$PAD680

In [21]:
# Complete t-test
t_test_result <- t.test(married_data, not_married_data)
print(t_test_result)


	Welch Two Sample t-test

data:  married_data and not_married_data
t = -3.8458, df = 7414, p-value = 0.0001212
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -28.121958  -9.132616
sample estimates:
mean of x mean of y 
 353.2871  371.9144 



The mean sedentary behavior time between those who are married is lower (353.29) compared to those who are not married (371.91). These results are significant (t= -3.846, p< 0.05) which means that married individuals spend less time sitting at school, at home, getting to and from places, or with friends including time spent sitting at a desk, traveling in a car or bus, reading, playing cards, watching television, or using a computer.

In [22]:
# Question 3: "How do age and marital status affect systolic blood pressure?"

In [23]:
# Load Blood Pressure - Oscillometric Measurements data
df_bp <- read_xpt("/content/BPXO_L.xpt")
head(df_bp)

SEQN   BPAOARM BPAOCSZ BPXOSY1 BPXODI1 BPXOSY2 BPXODI2 BPXOSY3 BPXODI3
1 130378 R       4       135     98      131      96     132      94    
2 130379 R       4       121     84      117      76     113      76    
3 130380 R       4       111     79      112      80     104      76    
4 130386 R       4       110     72      120      74     115      75    
5 130387 R       4       143     76      136      74     145      78    
6 130388 R       5       130     95      128     100     129     104    
  BPXOPLS1 BPXOPLS2 BPXOPLS3
1 82       79       82      
2 72       71       73      
3 84       83       77      
4 59       64       64      
5 80       80       77      
6 76       84       86

In [24]:
# Merge demographic and BP data
df_merged3<- merge(df_demo, df_bp, by="SEQN", all= TRUE)
head(df_merged3)

SEQN   SDDSRVYR RIDSTATR RIAGENDR RIDAGEYR RIDAGEMN RIDRETH1 RIDRETH3
1 130378 12       2        1        43       NA       5        6       
2 130379 12       2        1        66       NA       3        3       
3 130380 12       2        2        44       NA       2        2       
4 130381 12       2        2         5       NA       5        7       
5 130382 12       2        1         2       NA       3        3       
6 130383 12       1        2         3       NA       2        2       
  RIDEXMON RIDEXAGM ⋯ BPAOCSZ BPXOSY1 BPXODI1 BPXOSY2 BPXODI2 BPXOSY3 BPXODI3
1  2       NA       ⋯  4      135     98      131     96      132     94     
2  2       NA       ⋯  4      121     84      117     76      113     76     
3  1       NA       ⋯  4      111     79      112     80      104     76     
4  1       71       ⋯ NA       NA     NA       NA     NA       NA     NA     
5  2       34       ⋯ NA       NA     NA       NA     NA       NA     NA     
6 NA       NA       ⋯ NA       NA     NA       NA     NA       NA     NA     
  BPXOPLS1 BPXOPLS2 BPXOPLS3
1 82       79       82      
2 72       71       73      
3 84       83       77      
4 NA       NA       NA      
5 NA       NA       NA      
6 NA       NA       NA

In [25]:
names(df_merged3)

[1] "SEQN"     "SDDSRVYR" "RIDSTATR" "RIAGENDR" "RIDAGEYR" "RIDAGEMN"
 [7] "RIDRETH1" "RIDRETH3" "RIDEXMON" "RIDEXAGM" "DMQMILIZ" "DMDBORN4"
[13] "DMDYRUSR" "DMDEDUC2" "DMDMARTZ" "RIDEXPRG" "DMDHHSIZ" "DMDHRGND"
[19] "DMDHRAGZ" "DMDHREDZ" "DMDHRMAZ" "DMDHSEDZ" "WTINT2YR" "WTMEC2YR"
[25] "SDMVSTRA" "SDMVPSU"  "INDFMPIR" "BPAOARM"  "BPAOCSZ"  "BPXOSY1" 
[31] "BPXODI1"  "BPXOSY2"  "BPXODI2"  "BPXOSY3"  "BPXODI3"  "BPXOPLS1"
[37] "BPXOPLS2" "BPXOPLS3"

In [26]:
# Recode marital status
df_question3 <-df_merged3 %>%
  mutate(
    marital_status = ifelse(DMDMARTZ== 1, "Married", "Not Married")
  )
  head(df_question3)


SEQN   SDDSRVYR RIDSTATR RIAGENDR RIDAGEYR RIDAGEMN RIDRETH1 RIDRETH3
1 130378 12       2        1        43       NA       5        6       
2 130379 12       2        1        66       NA       3        3       
3 130380 12       2        2        44       NA       2        2       
4 130381 12       2        2         5       NA       5        7       
5 130382 12       2        1         2       NA       3        3       
6 130383 12       1        2         3       NA       2        2       
  RIDEXMON RIDEXAGM ⋯ BPXOSY1 BPXODI1 BPXOSY2 BPXODI2 BPXOSY3 BPXODI3 BPXOPLS1
1  2       NA       ⋯ 135     98      131     96      132     94      82      
2  2       NA       ⋯ 121     84      117     76      113     76      72      
3  1       NA       ⋯ 111     79      112     80      104     76      84      
4  1       71       ⋯  NA     NA       NA     NA       NA     NA      NA      
5  2       34       ⋯  NA     NA       NA     NA       NA     NA      NA      
6 NA       NA       ⋯  NA     NA       NA     NA       NA     NA      NA      
  BPXOPLS2 BPXOPLS3 marital_status
1 79       82       Married       
2 71       73       Married       
3 83       77       Married       
4 NA       NA       NA            
5 NA       NA       NA            
6 NA       NA       NA

In [27]:
# Create new table with only the columns that are needed
df_small3 <- df_question3[, c("RIDAGEYR", "marital_status", "BPXOSY3")]
head(df_small3)

RIDAGEYR marital_status BPXOSY3
1 43       Married        132    
2 66       Married        113    
3 44       Married        104    
4  5       NA              NA    
5  2       NA              NA    
6  3       NA              NA

In [28]:
# Drop missing data or invalid codes
df_clean3 <- na.omit(df_small3)
head(df_clean3)

RIDAGEYR marital_status BPXOSY3
1  43       Married        132    
2  66       Married        113    
3  44       Married        104    
9  34       Married        115    
10 68       Not Married    145    
11 27       Married        129

In [29]:
# Count the number of married people
table(df_clean3$marital_status)


    Married Not Married 
       3186        2656 

In [30]:
# BP difference between married and not married people
df_clean3 %>%
  group_by(marital_status) %>%
  summarise(average_BP = mean(BPXOSY3, na.rm = TRUE))

marital_status average_BP
1 Married        122.6099  
2 Not Married    122.8532

In [31]:
# Create two groups for t-test
married_bp <- subset(df_clean3, marital_status == "Married")$BPXOSY3
not_married_bp <- subset(df_clean3, marital_status == "Not Married")$BPXOSY3

In [32]:
# Complete t-test
t_test_result <- t.test(married_bp, not_married_bp, var.equal = FALSE)
print(t_test_result)


	Welch Two Sample t-test

data:  married_bp and not_married_bp
t = -0.50224, df = 5568.8, p-value = 0.6155
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.1930016  0.7063875
sample estimates:
mean of x mean of y 
 122.6099  122.8532 



The married individuals had a mean systolic blood pressure of 122.61 while unmarried had a mean systolic blood pressure of 122.85. The results indicated no statistically significant difference between the two groups,
(t = –0.50, p = 0.616). This shows us that marital status does not effect systolic blood pressure.

In [33]:
# Complete correlation test between age and BP
cor_test <- cor.test(df_clean3$RIDAGEYR, df_clean3$BPXOSY3, use = "complete.obs")
cor_test


	Pearson's product-moment correlation

data:  df_clean3$RIDAGEYR and df_clean3$BPXOSY3
t = 29.969, df = 5840, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.3426610 0.3871162
sample estimates:
      cor 
0.3650967 


A Pearson correlation showed a positive correlation between age and systolic blood pressure (r = 0.365, p < 0.001). This tells us that as age increases, systolic blood pressure increases.

In [34]:
# Question 4: Is there a correlation between self-reported weight and minutes of sedentary behavior?

In [35]:
# Load WHQ data for self-reported weight data WHD020
df_whq <-read_xpt("/content/WHQ_L.xpt")
df_whq

SEQN   WHD010 WHD020 WHD050 WHQ070
1    130378 71      190    200   1     
2    130379 70      220    220   2     
3    130380 60      150    165   1     
4    130384 68      204    212   1     
5    130385 70      240    240   2     
6    130386 68      200    180   2     
7    130387 67      215    215   2     
8    130388 66      270    300   1     
9    130389 67      175    185   1     
10   130390 64      277   9999   1     
11   130391 69      220    265   2     
12   130392 61      228    235   1     
13   130393 70      307    320   1     
14   130394 70      174    175   2     
15   130395 63      280    280   2     
16   130396 62      143    143   2     
17   130397 63      285    285   1     
18   130398 63      160    165   2     
19   130399 70      160    160   2     
20   130400 67     9999   9999   2     
21   130401 68      190    180   2     
22   130402 70      132    132   1     
23   130404 70      170    170   1     
24   130406 65      160    160   2     
25   130407 63      145    145   1     
26   130408 64      165    155   2     
27   130409 72      192    199   2     
28   130410 60      139    139   2     
29   130413 63      165    185   2     
30   130414 65      300    290   1     
⋮    ⋮      ⋮      ⋮      ⋮      ⋮     
8472 142268 71     165    170    2     
8473 142269 64     350    350    2     
8474 142270 74     185    210    2     
8475 142273 66     130    125    2     
8476 142278 66     183    179    2     
8477 142279 72     212    215    2     
8478 142280 69     250    270    1     
8479 142283 67     275    280    1     
8480 142284 74     205    202    2     
8481 142285 69     195    180    1     
8482 142286 70     180    175    1     
8483 142287 69     168    168    2     
8484 142288 64     125    125    2     
8485 142289 60     104    104    2     
8486 142290 72     228    240    1     
8487 142293 67     180    170    2     
8488 142294 62     163    170    1     
8489 142295 67     204    234    2     
8490 142297 67     158    158    2     
8491 142298 67     245    290    1     
8492 142299 72     248    300    1     
8493 142300 62     169    172    1     
8494 142301 65     180    180    1     
8495 142302 65     125    125    2     
8496 142303 63     165    160    1     
8497 142305 61     137    134    2     
8498 142307 61     206    235    1     
8499 142308 69     174    180    1     
8500 142309 69     200    170    2     
8501 142310 64     157    157    1

In [36]:
# Combine PAQ and WHQ data
df_merged4 <-merge(df_paq, df_whq, by = "SEQN", all = TRUE)
df_merged4

SEQN   PAD790Q PAD790U PAD800 PAD810Q PAD810U PAD820 PAD680 WHD010 WHD020
1    130378  3      W        45     3      W       45      360   71      190  
2    130379  4      W        45     3      W       45      480   70      220  
3    130380  1      W        20     0              NA      240   60      150  
4    130384  0               NA     0              NA       60   68      204  
5    130385  1      D        90     1      W       60      180   70      240  
6    130386  1      W        30     1      M       30      180   68      200  
7    130387  0               NA     0              NA     1200   67      215  
8    130388  5      W       180     0              NA      360   66      270  
9    130389  3      W        45     2      M       30      720   67      175  
10   130390  1      W        60     0              NA      300   64      277  
11   130391  3      D        10     1      D       20      240   69      220  
12   130392  0               NA     0              NA      300   61      228  
13   130393  4      W        45     1      W       60      720   70      307  
14   130394  7      W        45     4      W       30      420   70      174  
15   130395  3      W        15     0              NA      360   63      280  
16   130396  3      W        20     0              NA      240   62      143  
17   130397  0               NA     0              NA      480   63      285  
18   130398  7      W       120     2      Y       60       60   63      160  
19   130399  6      W        90     2      W       60      360   70      160  
20   130400  0               NA     0              NA      960   67     9999  
21   130401  0               NA     0              NA      960   68      190  
22   130402  1      W        60     1      M       60      840   70      132  
23   130404  5      W        50     3      W       30      120   70      170  
24   130406  5      W        60     3      W       60      300   65      160  
25   130407  5      W        50     1      W       40      180   63      145  
26   130408  1      D        45     3      W       45      120   64      165  
27   130409  3      W        60     1      M       60      300   72      192  
28   130410  0               NA     0              NA      120   60      139  
29   130413  5      W        50     0              NA       60   63      165  
30   130414 NA      NA       NA    NA      NA      NA       NA   65      300  
⋮    ⋮      ⋮       ⋮       ⋮      ⋮       ⋮       ⋮      ⋮      ⋮      ⋮     
8472 142268    6    W       360    1       W       120    300    71     165   
8473 142269    0             NA    0                NA    600    64     350   
8474 142270    0             NA    0                NA    960    74     185   
8475 142273    5    W        60    5       W        60    420    66     130   
8476 142278    1    D        30    0                NA    180    66     183   
8477 142279    2    W       150    0                NA    420    72     212   
8478 142280    1    W       120    1       W        60    180    69     250   
8479 142283 9999             NA    2       W        60    300    67     275   
8480 142284    1    D        60    3       W        60    360    74     205   
8481 142285    3    W        30    0                NA    300    69     195   
8482 142286    3    W       180    0                NA    420    70     180   
8483 142287    2    D        60    0                NA    300    69     168   
8484 142288    7    W        30    0                NA    360    64     125   
8485 142289    3    W        60    3       W        60    150    60     104   
8486 142290    1    D        60    2       W        60    240    72     228   
8487 142293    2    W        15    0                NA    300    67     180   
8488 142294    0             NA    0                NA    960    62     163   
8489 142295    0             NA    0                NA    480    67     204   
8490 142297    3    W        15    5       M        45 

In [37]:
# Keep columns PAD680 and WHD020
df_question4 <- df_merged4[, c("SEQN", "PAD680", "WHD020")]
df_question4

SEQN   PAD680 WHD020
1    130378  360    190  
2    130379  480    220  
3    130380  240    150  
4    130384   60    204  
5    130385  180    240  
6    130386  180    200  
7    130387 1200    215  
8    130388  360    270  
9    130389  720    175  
10   130390  300    277  
11   130391  240    220  
12   130392  300    228  
13   130393  720    307  
14   130394  420    174  
15   130395  360    280  
16   130396  240    143  
17   130397  480    285  
18   130398   60    160  
19   130399  360    160  
20   130400  960   9999  
21   130401  960    190  
22   130402  840    132  
23   130404  120    170  
24   130406  300    160  
25   130407  180    145  
26   130408  120    165  
27   130409  300    192  
28   130410  120    139  
29   130413   60    165  
30   130414   NA    300  
⋮    ⋮      ⋮      ⋮     
8472 142268 300    165   
8473 142269 600    350   
8474 142270 960    185   
8475 142273 420    130   
8476 142278 180    183   
8477 142279 420    212   
8478 142280 180    250   
8479 142283 300    275   
8480 142284 360    205   
8481 142285 300    195   
8482 142286 420    180   
8483 142287 300    168   
8484 142288 360    125   
8485 142289 150    104   
8486 142290 240    228   
8487 142293 300    180   
8488 142294 960    163   
8489 142295 480    204   
8490 142297 300    158   
8491 142298 480    245   
8492 142299 120    248   
8493 142300  90    169   
8494 142301 360    180   
8495 142302 120    125   
8496 142303 360    165   
8497 142305 480    137   
8498 142307 480    206   
8499 142308 600    174   
8500 142309 240    200   
8501 142310 360    157

In [38]:
# Replace invalid values (7777 and 9999) with NaN
df_question4$WHD020[df_question4$WHD020 %in% c(7777, 9999)] <- NA
df_question4$PAD680[df_question4$PAD680 %in% c(7777, 9999)] <- NA

In [39]:
# Drop missing data or invalid codes
df_clean4<- na.omit(df_question4)
head(df_clean4)

SEQN   PAD680 WHD020
1 130378 360    190   
2 130379 480    220   
3 130380 240    150   
4 130384  60    204   
5 130385 180    240   
6 130386 180    200

In [40]:
# Complete correlation test
cor_test <- cor.test(df_clean4$PAD680, df_clean4$WHD020, use = "complete.obs")
cor_test


	Pearson's product-moment correlation

data:  df_clean4$PAD680 and df_clean4$WHD020
t = 14.081, df = 7952, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.1344554 0.1773405
sample estimates:
      cor 
0.1559715 


There is a small but statistically significant correlation between self-reported weight and minutes of sedentary behavior as the Pearson correlation coefficient is 0.1560.

In [41]:
# Question 5: Do married women report greater feelings of tiredness or having little energy than unmarried women?

In [42]:
# Load DPQ data
df_dpq <- read_xpt("/content/DPQ_L.xpt")
df_dpq

SEQN   DPQ010 DPQ020 DPQ030 DPQ040 DPQ050 DPQ060 DPQ070 DPQ080 DPQ090
1    130378 NA     NA     NA     NA     NA     NA     NA     NA     NA    
2    130379  0      0      1      0      0      0      0      0      0    
3    130380  0      0      1      1      0      0      0      0      0    
4    130386  0      0      0      0      0      1      0      0      0    
5    130387  0      0      0      0      0      0      0      0      0    
6    130388 NA     NA     NA     NA     NA     NA     NA     NA     NA    
7    130389  0      0      0      0      0      0      0      0      0    
8    130390 NA     NA     NA     NA     NA     NA     NA     NA     NA    
9    130391  3      3      3      3      3      3      3      2      1    
10   130392  0      0      0      1      0      0      0      0      0    
11   130393  0      0      0      1      1      0      0      0      0    
12   130394  0      0      0      0      0      0      0      0      0    
13   130395  1      1      2      3      3      1      2      1      0    
14   130396  1      1      1      1      1      1      1      0      0    
15   130397  1      1      1      1      3      1      0      0      0    
16   130398  0      0      0      0      0      0      0      0      0    
17   130399  0      0      0      1      0      0      0      0      0    
18   130400  0      0      0      0      1      0      1      0      0    
19   130401  1      1      2      1      0      1      0      0      1    
20   130402 NA     NA     NA     NA     NA     NA     NA     NA     NA    
21   130404  3      0      0      0      0      0      1      0      0    
22   130406  0      0      0      0      1      0      1      0      0    
23   130407  0      0      0      0      0      0      0      0      0    
24   130408  3      3      2      2      2      3      2      3      2    
25   130409  0      0      0      0      0      0      0      0      0    
26   130413  1      1      3      1      1      0      1      2      0    
27   130415  2      1      1      2      3      0      3      0      0    
28   130416  0      0      1      0      0      0      0      0      0    
29   130418  1      1      2      3      0      2      0      0      0    
30   130419  2      2      2      3      1      3      3      0      1    
⋮    ⋮      ⋮      ⋮      ⋮      ⋮      ⋮      ⋮      ⋮      ⋮      ⋮     
6308 142263  0      0      0      1      0      0      0      0      0    
6309 142264  2      1      1      1      1      0      0      1      0    
6310 142266 NA     NA     NA     NA     NA     NA     NA     NA     NA    
6311 142267  0      0      3      2      0      0      0      0      0    
6312 142268  0      0      0      0      0      0      0      0      0    
6313 142269  1      3      3      2      3      3      1      1      3    
6314 142273  1      0      0      1      0      0      0      0      0    
6315 142278  1      0      1      1      1      1      0      0      0    
6316 142279  1      1      1      1      0      0      0      0      0    
6317 142280  3      2      2      2      2      0      3      0      2    
6318 142283  2      2      3      2      3      0      0      1      0    
6319 142284  0      0      1      1      0      0      0      0      0    
6320 142286  3      0      0      3      0      0      0      0      0    
6321 142287  3      0      0      0      0      0      0      0      0    
6322 142288  0      0      1      1      2      0      0      0      0    
6323 142289  0      0      1      0      0      0      0      0      0    
6324 142290  0      0      0      0      0      0      0      0      0    
6325 142293 NA     NA     NA     NA     NA     NA     NA     NA     NA    
6326 142295  0      0      1      1      0      0      0      0      0    
6327 142298  0      0      0      0      1      0      0      0      0    
6328 142299  0      0      0      0      0      0      0      0      0    
6329 142300  1      0      1  

In [43]:
# Merge DPQ and demo data
df_merged5 <-merge(df_demo, df_dpq, by = "SEQN", all = FALSE)
df_merged5

SEQN   SDDSRVYR RIDSTATR RIAGENDR RIDAGEYR RIDAGEMN RIDRETH1 RIDRETH3
1    130378 12       2        1        43       NA       5        6       
2    130379 12       2        1        66       NA       3        3       
3    130380 12       2        2        44       NA       2        2       
4    130386 12       2        1        34       NA       1        1       
5    130387 12       2        2        68       NA       3        3       
6    130388 12       2        2        27       NA       4        4       
7    130389 12       2        1        59       NA       3        3       
8    130390 12       2        2        31       NA       3        3       
9    130391 12       2        2        33       NA       3        3       
10   130392 12       2        2        74       NA       3        3       
11   130393 12       2        1        39       NA       3        3       
12   130394 12       2        1        51       NA       3        3       
13   130395 12       2        2        33       NA       5        7       
14   130396 12       2        2        56       NA       3        3       
15   130397 12       2        2        67       NA       3        3       
16   130398 12       2        1        47       NA       1        1       
17   130399 12       2        1        68       NA       3        3       
18   130400 12       2        1        26       NA       2        2       
19   130401 12       2        1        18       NA       3        3       
20   130402 12       2        2        26       NA       3        3       
21   130404 12       2        2        73       NA       3        3       
22   130406 12       2        2        26       NA       3        3       
23   130407 12       2        2        73       NA       3        3       
24   130408 12       2        1        80       NA       4        4       
25   130409 12       2        1        68       NA       3        3       
26   130413 12       2        2        67       NA       2        2       
27   130415 12       2        2        61       NA       4        4       
28   130416 12       2        2        59       NA       2        2       
29   130418 12       2        1        53       NA       3        3       
30   130419 12       2        2        18       NA       5        6       
⋮    ⋮      ⋮        ⋮        ⋮        ⋮        ⋮        ⋮        ⋮       
6308 142263 12       2        2        44       NA       3        3       
6309 142264 12       2        2        63       NA       2        2       
6310 142266 12       2        2        66       NA       4        4       
6311 142267 12       2        1        57       NA       3        3       
6312 142268 12       2        1        32       NA       4        4       
6313 142269 12       2        2        32       NA       4        4       
6314 142273 12       2        1        18       NA       5        6       
6315 142278 12       2        2        51       NA       5        7       
6316 142279 12       2        1        72       NA       3        3       
6317 142280 12       2        2        23       NA       5        7       
6318 142283 12       2        2        29       NA       5        7       
6319 142284 12       2        1        57       NA       3        3       
6320 142286 12       2        1        70       NA       3        3       
6321 142287 12       2        1        70       NA       5        7       
6322 142288 12       2        2        54       NA       3        3       
6323 142289 12       2        2        68       NA       3        3       
6324 142290 12       2        1        60       NA       3        3       
6325 142293 12       2        1        44       NA       2        2       
6326 142295 12       2        2        80       NA       3        3       
6327 142298 12       2        1        60       NA       5        7       
6328 142299 12       2        1        33       NA       1        1       
6329 142300 12       2        

In [44]:
# Keep columns RIAGENDR, DPQ040, DMDMARTZ
df_question5 <- df_merged5[, c("SEQN", "RIAGENDR", "DPQ040", "DMDMARTZ")]
df_question5

SEQN   RIAGENDR DPQ040 DMDMARTZ
1    130378 1        NA      1      
2    130379 1         0      1      
3    130380 2         1      1      
4    130386 1         0      1      
5    130387 2         0      3      
6    130388 2        NA      1      
7    130389 1         0      1      
8    130390 2        NA      1      
9    130391 2         3      3      
10   130392 2         1      1      
11   130393 1         1      1      
12   130394 1         0      1      
13   130395 2         3      3      
14   130396 2         1      1      
15   130397 2         1      1      
16   130398 1         0      1      
17   130399 1         1      1      
18   130400 1         0      3      
19   130401 1         1     NA      
20   130402 2        NA      3      
21   130404 2         0      2      
22   130406 2         0      1      
23   130407 2         0      2      
24   130408 1         2     99      
25   130409 1         0      1      
26   130413 2         1      1      
27   130415 2         2      1      
28   130416 2         0      2      
29   130418 1         3      1      
30   130419 2         3     NA      
⋮    ⋮      ⋮        ⋮      ⋮       
6308 142263 2         1      2      
6309 142264 2         1      1      
6310 142266 2        NA      2      
6311 142267 1         2      1      
6312 142268 1         0      1      
6313 142269 2         2      3      
6314 142273 1         1     NA      
6315 142278 2         1      2      
6316 142279 1         1      1      
6317 142280 2         2      3      
6318 142283 2         2      1      
6319 142284 1         1      1      
6320 142286 1         3      1      
6321 142287 1         0      1      
6322 142288 2         1      1      
6323 142289 2         0      1      
6324 142290 1         0      1      
6325 142293 1        NA      1      
6326 142295 2         1      2      
6327 142298 1         0      1      
6328 142299 1         0      1      
6329 142300 2         1      1      
6330 142301 2         1      2      
6331 142302 1         0      3      
6332 142303 2         0      2      
6333 142305 2        NA      2      
6334 142307 2         0      3      
6335 142308 1         0      1      
6336 142309 1         0      2      
6337 142310 2         1      1

In [45]:
# Recode marital status
df_recode5 <- df_question5 %>%
  mutate(
    marital_status = ifelse(DMDMARTZ == 1, "Married", "Not Married")
  )
  df_recode5

SEQN   RIAGENDR DPQ040 DMDMARTZ marital_status
1    130378 1        NA      1       Married       
2    130379 1         0      1       Married       
3    130380 2         1      1       Married       
4    130386 1         0      1       Married       
5    130387 2         0      3       Not Married   
6    130388 2        NA      1       Married       
7    130389 1         0      1       Married       
8    130390 2        NA      1       Married       
9    130391 2         3      3       Not Married   
10   130392 2         1      1       Married       
11   130393 1         1      1       Married       
12   130394 1         0      1       Married       
13   130395 2         3      3       Not Married   
14   130396 2         1      1       Married       
15   130397 2         1      1       Married       
16   130398 1         0      1       Married       
17   130399 1         1      1       Married       
18   130400 1         0      3       Not Married   
19   130401 1         1     NA       NA            
20   130402 2        NA      3       Not Married   
21   130404 2         0      2       Not Married   
22   130406 2         0      1       Married       
23   130407 2         0      2       Not Married   
24   130408 1         2     99       Not Married   
25   130409 1         0      1       Married       
26   130413 2         1      1       Married       
27   130415 2         2      1       Married       
28   130416 2         0      2       Not Married   
29   130418 1         3      1       Married       
30   130419 2         3     NA       NA            
⋮    ⋮      ⋮        ⋮      ⋮        ⋮             
6308 142263 2         1      2       Not Married   
6309 142264 2         1      1       Married       
6310 142266 2        NA      2       Not Married   
6311 142267 1         2      1       Married       
6312 142268 1         0      1       Married       
6313 142269 2         2      3       Not Married   
6314 142273 1         1     NA       NA            
6315 142278 2         1      2       Not Married   
6316 142279 1         1      1       Married       
6317 142280 2         2      3       Not Married   
6318 142283 2         2      1       Married       
6319 142284 1         1      1       Married       
6320 142286 1         3      1       Married       
6321 142287 1         0      1       Married       
6322 142288 2         1      1       Married       
6323 142289 2         0      1       Married       
6324 142290 1         0      1       Married       
6325 142293 1        NA      1       Married       
6326 142295 2         1      2       Not Married   
6327 142298 1         0      1       Married       
6328 142299 1         0      1       Married       
6329 142300 2         1      1       Married       
6330 142301 2         1      2       Not Married   
6331 142302 1         0      3       Not Married   
6332 142303 2         0      2       Not Married   
6333 142305 2        NA      2       Not Married   
6334 142307 2         0      3       Not Married   
6335 142308 1         0      1       Married       
6336 142309 1         0      2       Not Married   
6337 142310 2         1      1       Married

In [52]:
# Keep only rows where DPQ040 and marital_status are not missing
df_clean5 <- na.omit(df_recode5)
df_clean5

SEQN   RIAGENDR DPQ040 DMDMARTZ marital_status
2    130379 1        0       1       Married       
3    130380 2        1       1       Married       
4    130386 1        0       1       Married       
5    130387 2        0       3       Not Married   
7    130389 1        0       1       Married       
9    130391 2        3       3       Not Married   
10   130392 2        1       1       Married       
11   130393 1        1       1       Married       
12   130394 1        0       1       Married       
13   130395 2        3       3       Not Married   
14   130396 2        1       1       Married       
15   130397 2        1       1       Married       
16   130398 1        0       1       Married       
17   130399 1        1       1       Married       
18   130400 1        0       3       Not Married   
21   130404 2        0       2       Not Married   
22   130406 2        0       1       Married       
23   130407 2        0       2       Not Married   
24   130408 1        2      99       Not Married   
25   130409 1        0       1       Married       
26   130413 2        1       1       Married       
27   130415 2        2       1       Married       
28   130416 2        0       2       Not Married   
29   130418 1        3       1       Married       
32   130422 1        0       1       Married       
34   130425 2        2       1       Married       
35   130426 1        1       2       Not Married   
36   130430 2        1       3       Not Married   
38   130434 1        1       3       Not Married   
39   130436 1        0       1       Married       
⋮    ⋮      ⋮        ⋮      ⋮        ⋮             
6304 142253 2        1      2        Not Married   
6305 142255 2        1      1        Married       
6306 142258 2        2      2        Not Married   
6307 142259 2        1      1        Married       
6308 142263 2        1      2        Not Married   
6309 142264 2        1      1        Married       
6311 142267 1        2      1        Married       
6312 142268 1        0      1        Married       
6313 142269 2        2      3        Not Married   
6315 142278 2        1      2        Not Married   
6316 142279 1        1      1        Married       
6317 142280 2        2      3        Not Married   
6318 142283 2        2      1        Married       
6319 142284 1        1      1        Married       
6320 142286 1        3      1        Married       
6321 142287 1        0      1        Married       
6322 142288 2        1      1        Married       
6323 142289 2        0      1        Married       
6324 142290 1        0      1        Married       
6326 142295 2        1      2        Not Married   
6327 142298 1        0      1        Married       
6328 142299 1        0      1        Married       
6329 142300 2        1      1        Married       
6330 142301 2        1      2        Not Married   
6331 142302 1        0      3        Not Married   
6332 142303 2        0      2        Not Married   
6334 142307 2        0      3        Not Married   
6335 142308 1        0      1        Married       
6336 142309 1        0      2        Not Married   
6337 142310 2        1      1        Married

In [53]:
# Keep only women in our dataframe (women =2)
df_women <- df_clean5[df_clean5$RIAGENDR == 2, ]
df_women

SEQN   RIAGENDR DPQ040 DMDMARTZ marital_status
3    130380 2        1      1        Married       
5    130387 2        0      3        Not Married   
9    130391 2        3      3        Not Married   
10   130392 2        1      1        Married       
13   130395 2        3      3        Not Married   
14   130396 2        1      1        Married       
15   130397 2        1      1        Married       
21   130404 2        0      2        Not Married   
22   130406 2        0      1        Married       
23   130407 2        0      2        Not Married   
26   130413 2        1      1        Married       
27   130415 2        2      1        Married       
28   130416 2        0      2        Not Married   
34   130425 2        2      1        Married       
36   130430 2        1      3        Not Married   
40   130437 2        1      2        Not Married   
43   130441 2        1      1        Married       
46   130444 2        0      2        Not Married   
47   130449 2        2      1        Married       
50   130456 2        1      1        Married       
51   130457 2        3      3        Not Married   
52   130461 2        1      2        Not Married   
56   130473 2        1      1        Married       
57   130474 2        1      1        Married       
60   130478 2        2      1        Married       
61   130484 2        1      3        Not Married   
63   130491 2        2      1        Married       
65   130493 2        0      3        Not Married   
66   130494 2        0      1        Married       
69   130500 2        1      3        Not Married   
⋮    ⋮      ⋮        ⋮      ⋮        ⋮             
6279 142208 2        0      3        Not Married   
6281 142212 2        0      1        Married       
6282 142213 2        0      2        Not Married   
6285 142216 2        0      1        Married       
6287 142219 2        1      1        Married       
6289 142226 2        1      2        Not Married   
6290 142227 2        1      2        Not Married   
6293 142232 2        2      1        Married       
6295 142236 2        0      1        Married       
6296 142239 2        3      2        Not Married   
6298 142242 2        0      2        Not Married   
6300 142245 2        0      1        Married       
6304 142253 2        1      2        Not Married   
6305 142255 2        1      1        Married       
6306 142258 2        2      2        Not Married   
6307 142259 2        1      1        Married       
6308 142263 2        1      2        Not Married   
6309 142264 2        1      1        Married       
6313 142269 2        2      3        Not Married   
6315 142278 2        1      2        Not Married   
6317 142280 2        2      3        Not Married   
6318 142283 2        2      1        Married       
6322 142288 2        1      1        Married       
6323 142289 2        0      1        Married       
6326 142295 2        1      2        Not Married   
6329 142300 2        1      1        Married       
6330 142301 2        1      2        Not Married   
6332 142303 2        0      2        Not Married   
6334 142307 2        0      3        Not Married   
6337 142310 2        1      1        Married

In [54]:
# Show number of married vs unmarried women
table(df_women$marital_status)


    Married Not Married 
       1455        1418 

In [55]:
# Create two groups for t-test
married_women_score<- subset(df_women, marital_status == "Married")$DPQ040
not_married_score <- subset(df_women, marital_status == "Not Married")$DPQ040

In [56]:
# Mean DPQ040 scores for both groups
mean(married_women_score, na.rm = TRUE)
mean(not_married_score, na.rm = TRUE)

[1] 0.8591065

[1] 1.065585

In [57]:
# Complete t-test
t_test_result <- t.test(married_women_score, not_married_score, var.equal = FALSE)
print(t_test_result)


	Welch Two Sample t-test

data:  married_women_score and not_married_score
t = -5.7919, df = 2840.4, p-value = 7.724e-09
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.2763807 -0.1365769
sample estimates:
mean of x mean of y 
0.8591065 1.0655853 



An independent samples t-test showed a significant difference between the two groups (t = –5.79, p < 0.001).
Married women reported lower tiredness compared to unmarried women in this sample.
